sqlglot/optimizer/qualify_tables.py

```diff
@@ -46,6 +46,15 @@ def qualify_tables(
     db = exp.parse_identifier(db, dialect=dialect) if db else None
     catalog = exp.parse_identifier(catalog, dialect=dialect) if catalog else None
 
+    if isinstance(expression, exp.AlterTable):
+        for reference in expression.find_all(exp.Table):
+            if isinstance(reference.this, exp.Identifier):
+                if not reference.args.get("db"):
+                    reference.set("db", db)
+                if not reference.args.get("catalog") and reference.args.get("db"):
+                    reference.set("catalog", catalog)
```

Comment:
Should we factor this section out? it's the same as in L78-82.

Reply:
this is not the right approach

Reply:
i'm going to factor this out and use walk

Reply:
@z3z1ma can you change it to do this instead

```python

    for table in expression.find_all(exp.Table):
        if isinstance(table.this, exp.Identifier):
            if not table.args.get("db"):
                table.set("db", db)
            if not table.args.get("catalog") and table.args.get("db"):
                table.set("catalog", catalog)
```

delete the other sections

Reply:
move your test into qualify_tables.sql